meshio/vtu/_vtu.py

```diff
@@ -53,6 +54,93 @@ def _cells_from_data(connectivity, offsets, types, cell_data_raw):
         if meshio_type == "polygon" or meshio_type == "polyhedron":
             # Polygons and polyhedra have unknown and varying number of nodes per cell.
 
+            if meshio_type == "polyhedron":
+                # The data format for face-cells is:
+                # num_faces_cell_0,
+                #   num_nodes_face_0, node_ind_0, node_ind_1, ..
+                #   num_nodes_face_1, node_ind_0, node_ind_1, ..
+                #   ...
+                # num_faces_cell_1,
+                #   ...
+                #
+                # The faceoffsets describes the end of the face description for each
+                # cell, but this is 1-offset, so it is effectively the start of the
+                # next cell.
+                #
+                # See https://vtk.org/Wiki/VTK/Polyhedron_Support for more.
+
+                # There is a tacit assumption here that the cell-data keys 'faces'
+                # and 'faceoffsets' are reserved for this purpose.
+                # A .vtu file which uses 'faces' to specify the face-cell relation
+                # of a polyhedron, and also to give other type of information for
+                # other cells may lead to random behavior, at best a strange error,
+                # here.
+                # Whether such a file is permissible under the .vtu standard has not
+                # been checked, we will take our chances.
+                try:
+                    # pop removes the fields that were added to the raw cell data
+                    # in _organize_cells()
+                    faces = cell_data_raw.pop("faces")
+                    faceoffsets = cell_data_raw.pop("faceoffsets")
+                    # Switch faceoffsets to give start points, not end points
+                    faceoffsets = numpy.append(0, faceoffsets[:-1])
```

Comment:
These two lines should go in the `else` block of `try`-`except`.

Reply:
Not sure I agree:  the fields are popped on the first iteration, and then available on other iterations. 

Another matter is that the usage of try-except is artificial. It would arguably be better to pop the information before the loop over b (L52). This would add a bit of entropy for non-polyhedron meshes. Let me know what you prefer.

Reply:
> Not sure I agree: the fields are popped on the first iteration, and then available on other iterations.

Right.

> Another matter is that the usage of try-except is artificial. It would arguably be better to pop the information before the loop over b (L52). This would add a bit of entropy for non-polyhedron meshes. Let me know what you prefer.

Just do it before the loop, let's see what that looks like.

Reply:
I ended up implementing a version of your suggestion below, with first iteration (although I called the variable found_polyhedron_faces_before). If nothing else, that is explicit on when the variable will be used.